shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1250,19 +1436,88 @@ - (CGRect)bounds {
+  }
+
+  FlutterTextRange* range = [FlutterTextRange
+      rangeWithNSRange:fml::RangeForCharactersInRange(self.text, NSMakeRange(0, self.text.length))];
+  return [self closestPositionToPoint:point withinRange:range];
 }
 
 - (NSArray*)selectionRectsForRange:(UITextRange*)range {
-  // TODO(cbracken) Implement.
-  return @[];
+  NSAssert([range.start isKindOfClass:[FlutterTextPosition class]],
+           @"Expected a FlutterTextPosition for range.start (got %@).", [range.start class]);
+  NSAssert([range.end isKindOfClass:[FlutterTextPosition class]],
+           @"Expected a FlutterTextPosition for range.end (got %@).", [range.end class]);
+  NSUInteger start = ((FlutterTextPosition*)range.start).index;
+  NSUInteger end = ((FlutterTextPosition*)range.end).index;
+  NSMutableArray* rects = [[[NSMutableArray alloc] init] autorelease];
+  for (NSUInteger i = 0; i < [_selectionRects count]; i++) {
+    if (_selectionRects[i].position >= start && _selectionRects[i].position <= end) {
+      float width = _selectionRects[i].rect.size.width;
+      if (start == end) {
+        width = 0;
+      }
+      CGRect rect = CGRectMake(_selectionRects[i].rect.origin.x, _selectionRects[i].rect.origin.y,
+                               width, _selectionRects[i].rect.size.height);
+      FlutterTextSelectionRect* selectionRect = [FlutterTextSelectionRect
+          selectionRectWithRectAndInfo:rect
+                              position:_selectionRects[i].position
+                      writingDirection:UITextWritingDirectionNatural
+                         containsStart:(i == 0)
+                           containsEnd:(i == fml::RangeForCharactersInRange(
+                                                 self.text, NSMakeRange(0, self.text.length))
+                                                 .length)
+                            isVertical:NO];
+      [rects addObject:selectionRect];
+    }
+  }
+  return rects;
 }
 
 - (UITextPosition*)closestPositionToPoint:(CGPoint)point withinRange:(UITextRange*)range {
-  // TODO(cbracken) Implement.
-  return range.start;
+  NSAssert([range.start isKindOfClass:[FlutterTextPosition class]],
+           @"Expected a FlutterTextPosition for range.start (got %@).", [range.start class]);
+  NSAssert([range.end isKindOfClass:[FlutterTextPosition class]],
+           @"Expected a FlutterTextPosition for range.end (got %@).", [range.end class]);
+  NSUInteger start = ((FlutterTextPosition*)range.start).index;
+  NSUInteger end = ((FlutterTextPosition*)range.end).index;
+
+  NSUInteger _closestIndex = 0;
+  CGRect _closestRect = CGRectZero;
+  NSUInteger _closestPosition = 0;
+  for (NSUInteger i = 0; i < [_selectionRects count]; i++) {
+    NSUInteger position = _selectionRects[i].position;
+    if (position >= start && position <= end) {
+      BOOL isFirst = _closestIndex == 0;
+      if (isFirst || isPositionCloserToPoint(point, _selectionRects[i].rect, _closestRect, NO)) {
```

Comment:
nit: Adding a comment here would be helpful for the `NO` argument, `/*checkRightBoundary=*/ NO`